File: 1_code/Assets/Game/Common/TalkingDataManager.lua
-- 创建时间:2021-09-06

TalkingDataManager = {}
local M = TalkingDataManager
local is_on_off = true
local login_id
local login_qd

function M.Init()
	if not is_on_off then return end
	-- local function GetLocalLoginData(name)
	-- 	local path
	-- 	if AppDefine.IsEDITOR() then
	-- 		path = Application.dataPath .. "/" .. name .. ".txt"
	-- 	else
	-- 		path = AppDefine.LOCAL_DATA_PATH .. "/" .. name .. ".txt"
	-- 	end
	-- 	if File.Exists(path) then
	-- 		return File.ReadAllText(path)
	-- 	else
	-- 		return ""
	-- 	end
	-- end
	
	-- --获取登录数据
	-- local function GetLoginIDAndQD()
	-- 	-- 游客 微信 手机号
	-- 	local login_qd = {"phone", "wechat", "youke"}
	-- 	local login_id
	-- 	local cur_qd = nil
	-- 	for k,v in pairs(login_qd) do
	-- 		cur_qd = v
	-- 		login_id = GetLocalLoginData(v)
	-- 		if login_id and login_id ~= "" then
	-- 			break
	-- 		end
	-- 	end
	-- 	return login_id,cur_qd
	-- end
	-- login_id,login_qd = GetLoginIDAndQD()

end

function M.OnLogin()
	if not is_on_off then return end
	local userInfo = MainModel.UserInfo
	if userInfo then
		talkingDataMgr:Init(userInfo.user_id,"normal")
		talkingDataMgr:SetProfileName(userInfo.name)
		talkingDataMgr:SetProfileType(0)
		talkingDataMgr:SetGameServer("normal")
	end
end

function M.BeginMission(missitonName)
	if not is_on_off then return end
	talkingDataMgr:BeginMission(missitonName)
end

function M.CompleteMission(missitonName)
	if not is_on_off then return end
	talkingDataMgr:CompleteMission(missitonName)
end

function M.FailedMission(missitonName,failedCause)
	if not is_on_off then return end
	failedCause = failedCause or ""
	talkingDataMgr:FailedMission(missitonName,failedCause)
end
--[[
	请求充值
	data = {
		orderId = "订单号",
		iapId = "iapId",
		currencyAmount = 648,
		currencyType = "CNY",
		virtualCurrencyAmount = 10000,
		paymentType = "UnionPay"
	}
]]
function M.OnChargeRequest(data)
	if not is_on_off then return end
	if not data.orderId then
		dump("<color=red>ChargeRequest没有订单id</color>")
		return
	end
	data.iapId = data.iapId or "unknown"
	data.paymentType = data.paymentType or "unknown"
	talkingDataMgr:OnChargeRequest(
		data.orderId,
		data.iapId,
		data.currencyAmount,
		data.currencyType,
		data.virtualCurrencyAmount,
		data.paymentType)
end
function M.OnChargeSuccess(orderId)
	if not is_on_off then return end
	talkingDataMgr:OnChargeSuccess(orderId)
end

function M.OnReward(virtualCurrencyAmount,reason)
	if not is_on_off then return end
	talkingDataMgr:OnReward(virtualCurrencyAmount,reason)
end

--购买道具
function M.OnPurchase(item,itemNumber,priceInVirtualCurrency)
	if not is_on_off then return end
	talkingDataMgr:OnPurchase(item,itemNumber,priceInVirtualCurrency)
end

--使用道具
function M.OnUse(item,itemNumber)
	if not is_on_off then return end
	talkingDataMgr:OnUse(item,itemNumber)
end

File: 1_code/Assets/LuaFramework/Scripts/Manager/TalkingDataManager.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class TalkingDataManager : Manager
{
    private TDGAProfile profile;
    public void Init(string login_id,string cur_channel)
    {
        
        TalkingDataGA.BackgroundSessionEnabled();
        TalkingDataGA.OnStart("063B2F04C97F4CC9AEBF52708F54B4D4", cur_channel);
        profile = TDGAProfile.SetProfile("User" + login_id);
#if TDGA_PUSH
#if UNITY_IPHONE
        UnityEngine.iOS.NotificationServices.RegisterForNotifications(
            UnityEngine.iOS.NotificationType.Alert |
            UnityEngine.iOS.NotificationType.Badge |
            UnityEngine.iOS.NotificationType.Sound);
#endif
#endif
    }

    public string GetDeviceId()
    {
        return TalkingDataGA.GetDeviceId();
    } 

    //获取安卓机匿名id
    public string GetOAID()
    {
        return TalkingDataGA.GetOAID();
    }
    //设置玩家昵称
    public void SetProfileName(string name)
    {
        profile.SetProfileName(name);
    }

    //设置玩家类型
    public void SetProfileType(int profileType)
    {
        profile.SetProfileType((ProfileType)profileType);
    }

    //设置玩家等级
    public void SetLevel(int level)
    {
        profile.SetLevel(level);
    }

    //设置性别
    public void SetGender(int gender)
    {
        profile.SetGender((Gender)gender);
    }

    //设置年龄
    public void SetAge(int age)
    {
        profile.SetAge(age);
    }

    //设置区服
    public void SetGameServer(string serverName)
    {
        profile.SetGameServer(serverName);
    } 

    //开始任务(用于记录关卡等)
    public void BeginMission(string missionName)
    {
        TDGAMission.OnBegin(missionName);
    }

    //任务完成
    public void CompleteMission(string missionName)
    {
        TDGAMission.OnCompleted(missionName);
    }

    //任务失败(可传入失败原因)
    public void FailedMission(string missionName,string failedCause)
    {
        TDGAMission.OnFailed(missionName,failedCause);
    }

    /*
    orderId : 订单号
    iapId : 接入点id
    currencyAmount : 货币面值
    currencyType : 货币类型（人民币 美元）
    virtualCurrencyAmount : 虚拟货币价值
    paymentType ：支付方式
    */
    public void OnChargeRequest(string orderId, string iapId, double currencyAmount, string currencyType, double virtualCurrencyAmount, string paymentType)
    {
        TDGAVirtualCurrency.OnChargeRequest(orderId, iapId, currencyAmount, currencyType, virtualCurrencyAmount, paymentType);
    }

    //支付成功
    public void OnChargeSuccess(string orderId)
    {
        TDGAVirtualCurrency.OnChargeSuccess(orderId);
    }

    //奖励虚拟货币
    public void OnReward(double virtualCurrencyAmount,string reason)
    {
        TDGAVirtualCurrency.OnReward(virtualCurrencyAmount,reason);
    }

    //购买
    public void OnPurchase(string item,int itemNumber,double priceInVirtualCurrency)
    {
        TDGAItem.OnPurchase(item, itemNumber,priceInVirtualCurrency);
    }

    public void OnUse(string item,int itemNumber)
    {
        TDGAItem.OnUse(item,itemNumber);
    }

    //设置localtion（用途不明）
    public void SetLocation(double latitude, double longitude)
    {
        TalkingDataGA.SetLocation(latitude, longitude);
    }
}
